src/encoded/types/biosample.py

```diff
@@ -513,6 +513,30 @@ def summary(self, request,
         return construct_biosample_summary([biosample_dictionary],
                                            sentence_parts)
 
+    @calculated_property(schema={
+        "title": "Perturbed",
+        "description": "A flag to indicate whether the biosample has been perturbed with a treatment or genetic modification.",
+        "type": "boolean",
+        "notSubmittable": True,
+    })
+    def perturbed(
+        self,
+        request,
+        treatments=None,
+        genetic_modifications=None,
+        model_organism_donor_modifications=None
+    ):
+        applied_modifications = get_applied_modifications(
+            genetic_modifications, model_organism_donor_modifications)
+        is_perturbation = []
+        for modification in applied_modifications:
+            gm = request.embed(modification, '@@object')
+            is_perturbation.append(gm.get('perturbation'))
+        if treatments or True in is_perturbation:
+            return True
+        else:
+            return False
+
```

Comment:
`return any(treatments) or any(is_perturbation)`

Reply:
You might want to return True right away if there are treatments. Faster than always going through applied_modifications.

I think you can do this in one line but might be a little too condensed for clarity:

```
return any(treatments) or any(
    (
        request.embed(m, '@@object').get('perturbation', False)
        for m in applied_modifications
    )
)
```